[dom-gpu] Jenkinsfile UpdateEB

lucamar opened the pull request:
The Jenkinsfile `UpdateEB` is based on the previous `ExperimentalEB` project to create new toolchains and update the recipes in a new Cray Programming Environment: please have also a look at https://jenkins.cscs.ch/job/UpdateEB.

The Jira issues for failed updates should be submitted automatically using the Groovy method `failedJiraTask` defined in `util.groovy`: for the moment, the site jira-tds.cscs.ch is configured, but Stefano will enable access of `jenscscs` to the main CSCS Jira site soon, so we will be able to test it later. 
I would like to select the epic of a newly created task using the plugin Jira Steps https://plugins.jenkins.io/jira-steps, but I haven't managed yet. The plugin provides many functions and unfortunately I did not have time to go through all of them: therefore, any advice is appreciated.

A GitHub pull request for successful updates should be submitted automatically, following @henrique's script: the project will create a text file `submit.txt` with the path (relative to `$WORKSPACE/easybuild`) to the successful recipes and dependencies. I select the recipes for the pull request from the software folder, using the file in the `easybuild` subfolder, since the ones in the `reprod` subfolder expand the local variables.

I would like to test the project as soon as the main Jira site is configured, to see if it is working as expected.

teojgo commented on jenkins/JenkinsfileUpdateEB:
```diff
@@ -0,0 +1,375 @@
+                    /* define eb_flags and command to build software */
+                    def eb_flags = "--ignore-locks -r"
+                    def tc_flags = "--try-toolchain-version=${params.cdt_version}"
+                    //def build_command = architecture == "" ?
```
You dan delete this

lucamar replied:
I left it as a comment since we should use the compute nodes, as a general rule. 
However, since currently the build on `scratch` is dramatically slow, I switched to the login node.

henrique commented on jenkins/JenkinsfileUpdateEB:
```diff
@@ -0,0 +1,375 @@
+                        echo -e "\nFailed Updates: \${#failed[@]} \n\${failed[@]}"
+
+                        # return the number of failed updates
+                        exit \${#failed[@]}
```
Is Jenkins going to continue to the next steps if you return != 0 ?

lucamar replied:
Yes, the next stages `Jira` and `GitHub` should be executed anyway

lucamar commented:
It seems that the GitHub stage is skipped if the Jira stage fails:
https://jenkins.cscs.ch/blue/organizations/jenkins/UpdateEB/detail/UpdateEB/18/pipeline/33

lucamar commented:
The Jira issue is submitted on Jira TDS for the time being: 
https://jira-tds.cscs.ch/projects/UES/issues/UES-1009

teojgo commented:
> It seems that the GitHub stage is skipped if the Jira stage fails:
> https://jenkins.cscs.ch/blue/organizations/jenkins/UpdateEB/detail/UpdateEB/18/pipeline/33

@lucamar yes if the assertion is not satisfied then it seems that it stops everything.

lucamar commented:
I have changed a bit the Jira Stage to avoid the error when the failures are none. 
I am proposing to rename the production files using the suffix `-dom-gpu` instead of `-gpu-dom`, to make the automatic updated easier. 
I have also created the new empty files for `20.08` and updated the symbolic links `dom-gpu` and `dom-mc`: in this way, we will be able to fill the new production files with the successful recipes and let `ProductionEB` build them, while we can still use `UpdateEB` to test other builds.
Therefore, the two processes can run in parallel and we don't need to create the sotware stack directly with `UpdateEB`.
